nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Discover/DiscoveryWorker.cs

```diff
@@ -232,6 +236,30 @@ private static ImmutableArray<string> ExpandEntryPointsIntoProjects(IEnumerable<
                         filesToExpand.Push(project.AbsolutePath);
                     }
                 }
+                else if (extension == ".slnx")
+                {
+                    var projects = XElement
```

Comment:
Use the official `.slnx` parser added in [this PR](https://github.com/dotnet/msbuild/pull/10794) instead of direct XML loading; I'm not paid enough to edit this when the official parser does something unexpected in 3 years :)